### ISSUE TITLE
Mark as required

### ISSUE THREAD

**Issue by rightisleft** [weichx/cerialize#41]
It would be great if we could have a decorator that allowed use to mark a paramater as required during serialize/deserialize/auto 

This would allow us to ensure that the contract is properly met.

**Comment by weichx** [comment]
How would imagine this plays out?

```typescript
class Thing {
   @autoseralizeAs(String, true) thing1 : string;
   //or
   @autoserializeRequired thing2 : string;

}
```

or something else? I like the idea, I'm unsure off the top of my head what the interface for that would look like

**Comment by Marche86** [comment]
I prefer

`@autoseralizeAs(String, true) thing1 : string;`

Like java style

_@autoseralizeAs(columnDefinition="string", required=true)_

**Comment by rightisleft** [comment]
or even just a seperate annotation

@required
@autoserialize 


or require all and then opt out

@requireAll
export class Foo {

@autoserialize thing1
@autoserialize thing2
@autoserialize thing3
@autoserialize thing4

@optional
@autoserialize thing5
}

**Comment by rightisleft** [comment]
Because interfaces are compile time only in typescript - i see a lot of 'loose' declarations where data could or could not be missing depending on the source. 

Having auto runtime checking of all properties would be a huge win

**Comment by weichx** [comment]
Yeah I agree. I think I prefer the separate annotation. I'll see what I can do about that, or feel free to submit a PR since it will take me a few days before I have time to look at this seriously

**Comment by rightisleft** [comment]
Happy to test it when its ready! I definitely dont have the bandwidth

**Comment by hwaterke** [comment]
Any news on the proposal?
This would allow me to drop all my PropTypes (React) that I'm still keeping for the runtime checks

**Comment by weichx** [comment]
No no update yet, haven't had the time unfortunately. feel free to submit a PR though
